### ISSUE TITLE
consul-sidekick is outdated - migrate to consul-k8s?

### ISSUE THREAD

**Issue by rrichardson** [grafana/loki#2444]
On Kubernetes 1.18 servers,  it appears that the consul-sidekick cannot query its ReplicaSet because it was moved in the API. 
I have submitted a PR to correct this: https://github.com/weaveworks/consul-sidekick/pull/9 
I've also published that sidekick in dockerhub at `eigenrick/consul-sidekick`
And have verified that it works on k8s 1.18.* clusters and functions with Loki ingesters/distributors just fine. 

However, consul-sidekick has fallen out of fashion as the method for managing a consul cluster on Kubernetes. 
[consul-k8s](https://github.com/weaveworks/consul-k8s) is the new hotness.  Should we consider updating the deployment to use that?

**Comment by slim-bean** [comment]
Hi @rrichardson thanks for the issue and for the legwork to address it!! looks like you are a bit ahead of us on kubernetes versions and we haven't run into this issue yet.

We will need to read up on consul-k8s to see what the best next step is, not sure how quickly we will get back to this.

It sounds like you were able to unblock yourself for now though?

**Comment by rrichardson** [comment]
@slim-bean - Yes.  I'm not blocked on this.

**Comment by iamparamvs** [comment]
> [consul-k8s](https://github.com/weaveworks/consul-k8s) is the new hotness.

Is it just me or that URL is giving a 404 to everyone? @rrichardson @slim-bean

**Comment by rrichardson** [comment]
@iamparamvs -  Wow. I really botched that URL.. it's https://github.com/hashicorp/consul-k8s 

Hashicorp vs Weaveworks..  that sort of mixup is liable to anger someone.

**Comment by bboreham** [comment]
As an Open Source startup, getting mistaken for Hashicorp is the highest of praise 😇 .

I've had success using config based on https://github.com/kelseyhightower/consul-on-kubernetes

**Comment by james-callahan** [comment]
My consul-sidekick logs are full of 
> Cannot access ReplicaSet (loki/consul-6b6cd6677)

Is this the same issue? This is on a k8s 1.17 cluster. And [someone in slack reported the same error in 1.16](https://app.slack.com/client/T05675Y01/CEPJRLQNL/thread/CEPJRLQNL-1606706852.376800?cdn_fallback=2)

**Comment by bboreham** [comment]
Yes, very likely the same error. The beta API was disabled first in Kubernetes 1.16, but there was a flag to turn it back on again so some people would not hit it as a hard error until later.

Let me repeat that I recommend using StatefulSet and not using `consul-sidekick`, however I have merged @rrichardson's PR.

**Comment by JStickler** [comment]
Closing as stale.  This issue was opened against a version of Loki that is no longer supported.  If this issues still exists in more recent versions of Loki, please leave a comment and we can reopen the issue.
